src/frequenz/sdk/actor/_run_utils.py

```diff
@@ -0,0 +1,38 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Utility functions to run and synchronize the execution of actors."""
+
+
+import asyncio
+import logging
+from typing import Any
+
+from ._decorator import BaseActor
+
+log = logging.getLogger(__name__)
+
+
+async def run(*actors: Any) -> None:
+    """Await the completion of all actors.
+
+    Args:
+        actors: the actors to be awaited.
+
+    Raises:
+        AssertionError: if any of the actors is not an instance of BaseActor.
+    """
+    pending_tasks = []
+    # Check that each actor is an instance of BaseActor at runtime,
+    # due to the indirection created by the actor decorator.
+    for actor in actors:
+        assert isinstance(actor, BaseActor), f"{actor} is not an instance of BaseActor"
+        pending_tasks.append(asyncio.create_task(actor.join()))
+
+    # Currently the actor decorator manages the life-cycle of the actor tasks
+    while pending_tasks:
+        done_tasks, pending_tasks = await asyncio.wait(
+            pending_tasks, return_when=asyncio.FIRST_COMPLETED
+        )
+        log.debug("Actor task done: %s", done_tasks)
```

Comment:
I think you should check if the task ended with an exception and log an error in that case (same if it was cancelled).

```suggestion
    for actor in actors:
        assert isinstance(actor, BaseActor), f"{actor} is not an instance of BaseActor"
        pending_tasks.append(asyncio.create_task(actor.join())),
             name=str(actor)

    # Currently the actor decorator manages the life-cycle of the actor tasks
    while pending_tasks:
        done_tasks, pending_tasks = await asyncio.wait(
            pending_tasks, return_when=asyncio.FIRST_COMPLETED
        )
        # This should always be only one task, but we handle many for extra safety
        for task in done:
            # Cancellation needs to be checked first, otherwise the other methods could raise a CancelledError
            if task.cancelled():
	         log.info("The actor %s was cancelled", task.get_name())
            elif exception := task.exception():
	         log.error("The actor %s was finished due to an uncaught exception", task.get_name(), exc_info=exception)
	     else:
	         log.info("The actor %s finished normally", task.get_name())
```

Reply:
This is a great idea. We might want to set a name for the actor at some point given that in the this case we are assigning names like `<frequenz.sdk.actor._decorator.actor.<locals>.ActorClass object at 0x7f1648012d10>`, just to keep it in mind...
I'll add the suggested changes